core/contracts/Store.sol

```diff
@@ -0,0 +1,112 @@
+/*
+  Store implementation.
+ 
+  An implementation of StoreInterface with a fee per second and withdraw functions for the owner.
+*/
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./StoreInterface.sol";
+import "./Withdrawable.sol";
+import "./FixedPoint.sol";
+import "./MultiRole.sol";
+
+
+// An implementation of StoreInterface that can accept Oracle fees in ETH or any arbitrary ERC20 token.
+contract Store is StoreInterface, MultiRole {
+
+    using SafeMath for uint;
+    using FixedPoint for FixedPoint.Unsigned;
+    using FixedPoint for uint;
+
+    enum Roles {
+        Governance
+    }
+
+    uint private fixedOracleFeePerSecond; // Percentage of 10^18. E.g., 1e18 is 100% Oracle fee.
```

Comment:
Shouldn't this be a fixed point number since it's a percentage represented as a decimal (i.e. 0.5 => 50%)?

Reply:
Yeah that does make sense. I ran into some trouble making a global variable a part of a struct, but I can mess around with that more.